x/earn/keeper/grpc_query.go

```diff
@@ -266,8 +369,65 @@ func (s queryServer) getVaultAllDeposits(
 	}, nil
 }
 
-// getAccountAllDeposits returns deposits for all vaults for a specific account
-func (s queryServer) getAccountAllDeposits(
+// getBkavaVaultAllDeposits returns all deposits for the aggregated bkava vault
+func (s queryServer) getBkavaVaultAllDeposits(
+	ctx sdk.Context,
+	req *types.QueryDepositsRequest,
+) (*types.QueryDepositsResponse, error) {
+	deposits := []types.DepositResponse{}
+	store := prefix.NewStore(ctx.KVStore(s.keeper.key), types.VaultShareRecordKeyPrefix)
+
+	pageRes, err := query.FilteredPaginate(
+		store,
+		req.Pagination,
+		func(key []byte, value []byte, accumulate bool) (bool, error) {
+			var record types.VaultShareRecord
+			err := s.keeper.cdc.Unmarshal(value, &record)
+			if err != nil {
+				return false, err
+			}
+
+			// Only those that have bkava deposits
+			if !vaultSharesContainBkava(record.Shares) {
+				// inform paginate that there was no match on this key
+				return false, nil
+			}
+
+			if accumulate {
+				// Get total value for all bkava
+				totalAccountValue, err := getAccountTotalValue(ctx, s.keeper, record.Depositor, record.Shares)
+				if err != nil {
+					return false, err
+				}
+
+				bkavaTotal := getTotalBkava(totalAccountValue)
+
+				// only add to results if paginate tells us to
+				deposits = append(deposits, types.DepositResponse{
+					Depositor: record.Depositor.String(),
+					// Only the specified shares of requested denom
+					Shares: nil,
+					Value:  sdk.NewCoins(bkavaTotal),
+				})
+			}
+
+			// inform paginate that were was a match on this key
+			return true, nil
+		},
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryDepositsResponse{
+		Deposits:   deposits,
+		Pagination: pageRes,
+	}, nil
+}
+
+// get1AccountAllDeposits returns deposits for all vaults for a specific account
+func (s queryServer) get1AccountAllDeposits(
```

Comment:
the number in the func is a bit hard to see. Thoughts on changing it to `getOneAccountAllDeposits` or maybe just `getAccountAllDeposits`?